engine/crates/runtime-local/src/rate_limiting/in_memory/key_based.rs

```diff
@@ -34,48 +37,73 @@ impl RateLimiterContext for RateLimitingContext {
     }
 }
 
-#[derive(Default)]
 pub struct InMemoryRateLimiter {
-    inner: HashMap<String, governor::DefaultKeyedRateLimiter<usize>>,
+    limiters: Arc<RwLock<HashMap<String, governor::DefaultKeyedRateLimiter<usize>>>>,
 }
 
 impl InMemoryRateLimiter {
-    pub fn runtime(config: KeyedRateLimitConfig<'_>) -> RateLimiter {
-        let mut limiter = Self::default();
+    pub fn runtime(
+        config: KeyedRateLimitConfig,
+        mut updates: mpsc::Receiver<HashMap<String, GraphRateLimit>>,
+    ) -> RateLimiter {
+        let mut limiters = HashMap::new();
 
         // add subgraph rate limiting configuration
-        for (name, rate_limit_config) in config.rate_limiting_configs {
-            limiter = limiter.with_rate_limiter(name, rate_limit_config);
+        for (name, config) in config.rate_limiting_configs {
+            let Some(limiter) = create_limiter(config) else {
+                continue;
+            };
+
+            limiters.insert(name.to_string(), limiter);
         }
 
-        RateLimiter::new(limiter)
-    }
+        let limiters = Arc::new(RwLock::new(limiters));
+        let limiters_copy = limiters.clone();
```

Comment:
Did you consider using a weak clone instead? Would avoid leaking futures if multiple RateLimiter are created (schema reload in cli/gateway, etc.)

Reply:
mm, yeah I don't like weaks that much but here that kind of makes sense.